### PR TITLE
Add feature/addition_of_sand_gravel_and_cement_gathering

### PR DESCRIPTION
**Pull request by KasperOnPC**
Please see #90 

Considerations:
- This uses existing assets, all but for the gravel which is a simple material change (no texture change).
- I have added 3 debug chains to debug, one for each resource.
- Each bag inventory item weighs 20kg and affects the character's stamina significantly.
- Players need a backpack to fit the item and there is no way to put it in their hands at the moment (someone could figure out other stances via animation and easily add this to each bag).
- Added filesize is ~11kb

Once this is in along with processing I'll be sure to update the chains with a processor and again once we have a money system enabling selling.

Thanks in advance for your time reviewing this :)

### CODE REVIEW

File: src/Worlds/DebugWorld/DebugWorld_GravelChain.layer [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  ActionsManagerComponent "{598A74222DFDE823}" {
+   ActionContexts {
+    UserActionContext "{598A74222DFDE81E}" {
+    }
+   }
+  }
+  RplComponent "{598A74222DFDE817}" {
+  }
+ }
```
Comment (Arkensor) [inline src/Worlds/DebugWorld/DebugWorld_GravelChain.layer:24]: I am not sure why the map saved all of this info. If we just used the prefab it should have just referenced the gravel pile prefab and listed none of the components.

**Review by Arkensor — CHANGES_REQUESTED** [review]
Good addition, only minor questions/remarks

- Where on the map can we find sand, gravel, and cement piles? I was thinking if we should override the prefab to add the action to all of them and server owners can put additional piles to designated areas where people are sure to find some? If they are common on the map this would however become a balancing problem.
- Does it maybe makes sense to make a re-useable base prefab for the bags to configure something like in-hand animations or do we simply keep them as inventory-only items? 
- `Gravel_bag` -> `GravelBag` & `Cement_bag` -> `CementBag` to keep the names consistent

**Comment by KasperOnPC** [comment]
> * Where on the map can we find sand, gravel, and cement piles? I was thinking if we should override the prefab to add the action to all of them and server owners can put additional piles to designated areas where people are sure to find some? If they are common on the map this would however become a balancing problem.

You can find these in abundance across the map. The idea here was that rather than overriding all of the existing to simply make these ones so server owners can place them down wherever they want them to be gathered from and make it obvious with map markers/signage/etc. It would certainly create an issue if it was a simple override, for example, there is a few of these piles at the "Concrete Processor" that would just make it so people could gather them without having to go anywhere. 

> * Does it maybe makes sense to make a re-useable base prefab for the bags to configure something like in-hand animations or do we simply keep them as inventory-only items?

I would love someone to build an in-hand animation for it, but doesn't seem like something that's going to be done in the medium term, so I would prefer to keep them as so for now. 

> * `Gravel_bag` -> `GravelBag` & `Cement_bag` -> `CementBag` to keep the names consistent

Complete

**Comment by Arkensor** [comment]
> "Concrete Processor" that would just make it so people could gather them without having to go anywhere.

I see. I can see why you did not go for an override then. My "problem" with this stuff is that as a player if I see the same sand piles somewhere that I can interact with at a different location that breaks the immersion a bit. So if it is possible to make all of them interactable no matter where they are (even if it makes little sense to do it there) that is super cool. One idea I had is, what if we intentionally use the fact that the sand is nearby the factory? Why not make it a super cheap and heavy resource that makes no sense to gather 10 km away just to drive it to the factory. I think the idea of people "farming" in the backyard is essentially doing a factory worker job on site. Might not give a lot of money but would allow us to let the piles of whatever be interactable there. 

And if someone is gathering sand we could make it that any other production chain that uses it is far away with a closer source of sand, so people do not go to random sand spawns to "cut the route short".

Artificially blocking the interaction with vanilla world items of the same type is of course always an option, but if we can design around it I think it would be a much more pleasant experience for people.

Where exactly is this plant located? Can you give me the coordinates? Word view -> Camera button right above the view box, right-click on the coordinates to copy the value.

**Comment by KasperOnPC** [comment]
>So if it is possible to make all of them interactable no matter where they are (even if it makes little sense to do it there) that is super cool. One idea I had is, what if we intentionally use the fact that the sand is nearby the factory? Why not make it a super cheap and heavy resource that makes no sense to gather 10 km away just to drive it to the factory. I think the idea of people "farming" in the backyard is essentially doing a factory worker job on site. Might not give a lot of money but would allow us to let the piles of whatever be interactable there.

I really like the idea of a Factory Job where you can remain on-site it's pretty cool. Currently, the site just has Sand & Gravel piles and not the "Cement Pile" so you could require players to go and collect cement from another location or just drop the cement piles at the Cement Processor as well to make it a Factory Job. 

I do think doing it as an override does remove some of the flexibility for server owners to be able to place all gathering sites wherever they want though. It would also reduce our ability to make use of existing assets in general if we set this precedent. 

For example, if we want to use the existing Cocoa Trees, overriding it would make it super easy to collect cocaine wherever you want. And if that wasn't the case it might cause confusion on why they can gather a resource from all the assets for some things but not others. 

> And if someone is gathering sand we could make it that any other production chain that uses it is far away with a closer source of sand, so people do not go to random sand spawns to "cut the route short".

Yeah, I don't mind this idea for a glass processor just being pretty far away from any source of sand to balance it. There are 8 Sand Piles across the map. 

But, one of the Sand Piles, for example, is just in a backyard. If we take this approach (of overriding every asset) it may limit where server owners can realistically place their "Glass Processors". If they place it wrong and don't think about the existing assets they could have 10 trucks pulling up at some random house in suburbia to load up on the sand.
image

I think this point combined with the point made above shows this might not be a great idea if we are trying to give Server Owners the most options when it comes to how they want to set this up. Because A) they won't be able to choose where the gathering happens and B) they will be restricted when it comes to where they can place the processor. 

> Artificially blocking the interaction with vanilla world items of the same type is of course always an option, but if we can design around it I think it would be a much more pleasant experience for people.

Yeah, I don't like the idea of blocking interactions. It would just seem buggy for Server Owners. 

> Where exactly is this plant located? Can you give me the coordinates? Word view -> Camera button right above the view box, right-click on the coordinates to copy the value.

Coordinates are: 
5076.32 54.58 10868.49

You can also search "ConcretePlant" on the map and it will come up. 


Keen for your and everyone else's thoughts on this! :)

**Comment by Arkensor** [comment]
> But, one of the Sand Piles, for example, is just in a backyard. If we take this approach (of overriding every asset) it may limit where server owners can realistically place their "Glass Processors". If they place it wrong and don't think about the existing assets they could have 10 trucks pulling up at some random house in suburbia to load up on the sand.

I understand, but I think this should be intended. If we have a resource that is practically everywhere on the map then that might not be a good idea for a high-value resource, but something like sand, for example, could just really be where ever you see it. Same as we do with apples right now. One has to keep in mind that if you want to avoid the usage of existing map objects that share the same prefab we could just have server owners remove balance destroying objects with a map utility to place down near it. 

What is the class name of the coca tree? I want to check how many there are and where they are placed.

**Pull request closed without merging**